### PR TITLE
Update permissions on Android, fix exception while uploading informed consent

### PR DESCRIPTION
**Pull request by bardram**
Fix of #306

### CODE REVIEW

File: lib/data/carp_backend.dart [hunk 1]
```diff
@@ -151,25 +151,36 @@ class CarpBackend {
...
+    Map<String, dynamic> json = signedConsent.toJson();
 
-    ConsentDocument? document;
+    ConsentDocument? uploadedConsent;
     try {
-      document = await CarpService().createConsentDocument(informedConsent);
+      uploadedConsent = await CarpService().createConsentDocument(json);
       info(
-          'Informed consent document uploaded successfully - id: ${document.id}');
-      bloc.hasInformedConsentBeenAccepted = true;
```
Comment (SlimShadyIAm) [inline lib/data/carp_backend.dart:167]: I can see that this has been removed, is this no longer needed?

Comment (SlimShadyIAm) [reply lib/data/carp_backend.dart:167]: Updating `bloc` on both success and failure ^

Comment (bardram) [reply lib/data/carp_backend.dart:167]: No. This has been moved to the bloc in order to keep the semantics of this method clear (it's only responsibility is to upload and not to mark anything locally in the app's BLoC). This is now handled in the bloc - see

https://github.com/cph-cachet/carp_studies_app/blob/4b2de58f0f4f7c2ef2dbf9a312ef4c50d164b966/lib/blocs/app_bloc.dart#L264

File: pubspec.lock [hunk 2]
```diff
@@ -266,21 +266,20 @@ packages:
...
     description:
-      name: carp_mobile_sensing
-      sha256: "8d371939d46e91bd96141fc558a1f35f64d81e3555176250efd39a25d0ca1d19"
-      url: "https://pub.dev"
-    source: hosted
-    version: "1.7.2"
+      path: "../carp/carp.sensing-flutter/carp_mobile_sensing"
+      relative: true
+    source: path
+    version: "1.8.1"
```
Comment (Panosfunk) [inline pubspec.lock:274]: This needs to be fixed before merge

Comment (bardram) [reply pubspec.lock:274]: This should not be an issues. The pubspec.lock file is generated from the pubspec file when the app is build. Actually, there has been some discussion previously why the lock file isn't gitignored in the first place. But Lars found some recommendation somewhere that lock files should be committed for APPS whereas it should be ignored for PACKAGES.

Comment (bardram) [reply pubspec.lock]: ... but I've committed the lock file now, and it should be pointing to pub.dev

**Review by SlimShadyIAm — APPROVED** [review]

**Pull request merged**